test-kotlin/src/com/intellij/rt/coverage/kotlin/runner.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2000-2020 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intellij.rt.coverage.kotlin
+
+import com.intellij.rt.coverage.data.ClassData
+import com.intellij.rt.coverage.data.LineData
+import com.intellij.rt.coverage.data.ProjectData
+import com.intellij.rt.coverage.util.FileUtil
+import com.intellij.rt.coverage.util.ProcessUtil
+import com.intellij.rt.coverage.util.ProjectDataLoader
+import com.intellij.rt.coverage.util.ResourceUtil
+import junit.framework.TestCase.assertEquals
+import java.io.File
+
+
+enum class LineStatus {
+    NONE, PARTIAL, FULL
```

Comment:
there is a common interface `com.intellij.rt.coverage.data.LineCoverage` with statuses. If you strongly prefer enums (I am ok, if you check that introducing them won't increase the size of the stored data), then please update existing interface; otherwise please use it as is

Reply:
There is a small overhead in using enum. 
Now a byte to string transformation is used to make test failure readable.